### PR TITLE
Double management

### PR DESCRIPTION
**Pull request by snopoke**
Spec: https://docs.google.com/a/dimagi.com/document/d/15gaQpzDm4UeiSz8BIIZPGA8a6xTR0T8LzxSNNClVsec/edit#

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_suite.py [hunk 1]
```diff
@@ -289,6 +289,39 @@ def test_graphing(self):
     def test_fixtures_in_graph(self):
         self._test_generic_suite('app_fixture_graphing', 'suite-fixture-graphing')
 
+    def _prep_case_list_form_app(self):
+        app = Application.wrap(self.get_json('app'))
+        case_module = app.get_module(0)
+        form = case_module.get_form(0)
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_suite.py]: F841 local variable 'form' is assigned to but never used

File: corehq/apps/app_manager/xform.py [hunk 2]
```diff
@@ -1156,17 +1157,26 @@ def make_delegation_stub_case_block():
...
 
-                if form.get_module().task_list.show:
+                if module.task_list.show:
                     delegation_case_block = make_delegation_stub_case_block()
 
             if 'open_case' in actions:
                 open_case_action = actions['open_case']
+                case_id = 'uuid()'
+                case_list_modules = (
+                    mod for mod in form.get_app().get_modules() if mod.case_list_form.form_id == form.get_unique_id()
```
Comment (dimagimon) [inline corehq/apps/app_manager/xform.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/app_manager/models.py [hunk 3]
```diff
@@ -789,9 +793,18 @@ def get_auto_gps_capture(self):
...
             return False
 
+    def is_registration_form(self, case_type=None):
+        """
+        Should return True if this form passes the following tests:
+         * does not require a case
+         * registers a case of type 'case_type' if supplied
+        """
+        raise NotImplementedError()
+
```
Comment (dimagimon) [inline corehq/apps/app_manager/models.py]: W293 blank line contains whitespace

**Comment by czue** [comment]
wow i did not realize this wasn't merged yet

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/app_manager/models.py, line 823 - E271 multiple spaces after keyword
* corehq/apps/app_manager/models.py, line 2138 - E265 block comment should start with '# '
* corehq/apps/app_manager/models.py, line 2139 - E265 block comment should start with '# '
* corehq/apps/app_manager/suite_xml.py, line 716 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/suite_xml.py, line 961 - E501 line too long (118 > 115 characters)
* corehq/apps/app_manager/suite_xml.py, line 1414 - E501 line too long (116 > 115 characters)
* corehq/apps/app_manager/tests/test_suite.py, line 393 - E303 too many blank lines (2)

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/app_manager/models.py, line 823 - E271 multiple spaces after keyword
* corehq/apps/app_manager/models.py, line 2138 - E265 block comment should start with '# '
* corehq/apps/app_manager/models.py, line 2139 - E265 block comment should start with '# '
* corehq/apps/app_manager/suite_xml.py, line 716 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/suite_xml.py, line 963 - E501 line too long (118 > 115 characters)
* corehq/apps/app_manager/suite_xml.py, line 1416 - E501 line too long (116 > 115 characters)
* corehq/apps/app_manager/tests/test_suite.py, line 393 - E303 too many blank lines (2)

**Comment by snopoke** [comment]
@czue, yea I know - it's weighing me down but hopefully should be done soon

File: corehq/apps/app_manager/models.py [hunk 4]
```diff
@@ -789,9 +801,31 @@ def get_auto_gps_capture(self):
...
         pass
 
+    @property
+    @memoized
+    def case_list_module(self):
+        case_list_modules = [
+            mod for mod in self.get_app().get_modules() if mod.case_list_form.form_id == self.unique_id
+        ]
+        assert len(case_list_modules) <= 1, "Form referenced my multiple modules"
+        return case_list_modules[0] if case_list_modules else  None
```
Comment (dimagimon) [inline corehq/apps/app_manager/models.py]: E271 multiple spaces after keyword

File: corehq/apps/app_manager/suite_xml.py [hunk 5]
```diff
@@ -701,6 +711,26 @@ def generic_fixture_instances(instance_name):
     return Instance(id=instance_name, src='jr://fixture/{}'.format(instance_name))
 
 
+def get_case_auto_select_xpath(form, action):
+    from corehq.apps.app_manager.models import AUTO_SELECT_USER, AUTO_SELECT_CASE, \
+            AUTO_SELECT_FIXTURE, AUTO_SELECT_RAW
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml.py]: E126 continuation line over-indented for hanging indent

File: corehq/apps/app_manager/models.py [hunk 6]
```diff
@@ -60,7 +60,8 @@
 from corehq.apps.domain.models import cached_property
 from corehq.apps.app_manager import current_builds, app_strings, remote_app
 from corehq.apps.app_manager import fixtures, suite_xml, commcare_settings
-from corehq.apps.app_manager.util import split_path, save_xform, get_correct_app_class, ParentCasePropertyBuilder
+from corehq.apps.app_manager.util import split_path, save_xform, get_correct_app_class, ParentCasePropertyBuilder, \
```
Comment (dimagimon) [inline corehq/apps/app_manager/models.py]: F401 'dict_get_path' imported but unused

Comment (dimagimon) [inline corehq/apps/app_manager/models.py]: F401 'dict_set_path' imported but unused

File: corehq/apps/app_manager/tests/test_suite.py [hunk 7]
```diff
@@ -304,6 +304,82 @@ def test_graphing(self):
...
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_suite.py]: F841 local variable 'form' is assigned to but never used

File: requirements/requirements.txt [hunk 8]
```diff
@@ -72,6 +72,7 @@ django-compressor==1.4
 django-angular==0.7.8
 Pycco==0.3.0
 django-mptt==0.6.1
+jsonpath-rw==1.3.0
```
Comment (snopoke) [inline requirements/requirements.txt]: @czue FYI in case you want to use it in UCR

Comment (czue) [inline requirements/requirements.txt]: :+1: thanks for the heads up

**Comment by snopoke** [comment]
Closing in favour of #5603

**Pull request closed without merging**